### PR TITLE
support skipper RouteGroups 

### PR DESCRIPTION
**Pull request by szuecs**
support skipper RouteGroups and create shared ALBs/NLBs and all annotations from ingress also in routegroups

TBD: change in client.patch() to change from `application/strategic-merge-patch+json` to  `application/merge-patch+json`, because the other is not supported for custom CRDs.

Signed-off-by: Sandor Szücs <sandor.szuecs@zalando.de>

### CODE REVIEW

File: kubernetes/adapter.go [hunk 1]
```diff
@@ -65,6 +68,7 @@ type Ingress struct {
 	SSLPolicy        string
 	IPAddressType    string
 	LoadBalancerType string
+	typ              string // "ingress" or "routegroup" to patch status based on that
```
Comment (mikkeloscar) [inline kubernetes/adapter.go]: maybe call it `resourceType` instead of this weird word? :)

**Comment by mikkeloscar** [comment]
Generally it seems fine to me. Will wait for the patch change.

File: kubernetes/adapter.go (hunk 1 above)
Comment (szuecs) [reply kubernetes/adapter.go]: fixed

Comment (szuecs) [reply kubernetes/adapter.go]: I changed the name from typ to resourceType.

**Comment by szuecs** [comment]
I tested that the patch version is fine also for changing ALBs. So we do not get leftovers or such things in the status.

**Comment by mikkeloscar** [comment]
:+1:

**Comment by szuecs** [comment]
:+1:

**Pull request merged**